### ISSUE TITLE
BUG: dhg.Hypergraph.from_feature_kNN() need .detach() for input feature tensor

### ISSUE THREAD

**Issue by realxuyang** [iMoonLab/DeepHypergraph#19]
![4dd0c21372c66101ad34fbe3c81360d](https://user-images.githubusercontent.com/45526497/198297759-113b653d-d976-4d64-91c4-ad446b4ba8f1.jpg)
![0bab6bd2905a774009a557aab8c51d8](https://user-images.githubusercontent.com/45526497/198297781-99108c6d-d015-4608-838f-cd30681ca961.jpg)

**Comment by yifanfeng97** [comment]
Got it! We will invoke .detach() for the input feature matrix.

**Comment by yifanfeng97** [comment]
Fix in ad3d7202e999dc02d0193308ffbb5797f1f0f11f
